ci: dpclang workflow started to fail on build stage on scheduling events

See (8/2/2026): https://github.com/intel/torch-xpu-ops/actions/runs/30742312302

[dpclang workflow](https://github.com/intel/torch-xpu-ops/blob/main/.github/workflows/pull_dpclang.yml) started to fail at some point on scheduling events during build stage. History of execution:

| Date | Status (build stage) |
| --- | --- |
| 8/2/2026 | failed |
| 7/26/2026 | failed |
| 7/19/2026 | passed | 

Error:
```
+ [[ '' == *\h\t\t\p\s\:\/\/* ]]
+ PYTORCH_REPO=https://github.com/pytorch/pytorch.git
+ PYTORCH_COMMIT=
+ [[ main == *\h\t\t\p\s\:\/\/* ]]
+ TORCH_XPU_OPS_REPO=https://github.com/intel/torch-xpu-ops.git
+ TORCH_XPU_OPS_COMMIT=main
+ python /home/jenkins/actions-runner/_work/torch-xpu-ops/torch-xpu-ops/torch-xpu-ops/.github/scripts/prepare_pytorch.py --workspace=/home/jenkins/actions-runner/_work/torch-xpu-ops/torch-xpu-ops --pytorch-repo=https://github.com/pytorch/pytorch.git --pytorch-commit= --torch-xpu-ops-repo=https://github.com/intel/torch-xpu-ops.git --torch-xpu-ops-commit=main --onednn-commit=
+ git clone https://github.com/pytorch/pytorch.git /home/jenkins/actions-runner/_work/torch-xpu-ops/torch-xpu-ops/pytorch
Cloning into '/home/jenkins/actions-runner/_work/torch-xpu-ops/torch-xpu-ops/pytorch'...
Updating files:  76% (16616/21708)
...
Updating files: 100% (21708/21708), done.
+ git checkout 
fatal: empty string is not a valid pathspec. please use . instead if you meant to match all paths
Traceback (most recent call last):
  File "/home/jenkins/actions-runner/_work/torch-xpu-ops/torch-xpu-ops/torch-xpu-ops/.github/scripts/prepare_pytorch.py", line 159, in <module>
    main()
  File "/home/jenkins/actions-runner/_work/torch-xpu-ops/torch-xpu-ops/torch-xpu-ops/.github/scripts/prepare_pytorch.py", line 83, in main
    git("checkout", args.pytorch_commit)
  File "/home/jenkins/actions-runner/_work/torch-xpu-ops/torch-xpu-ops/torch-xpu-ops/.github/scripts/prepare_pytorch.py", line 37, in git
    run(["git", *args], **kwargs)
  File "/home/jenkins/actions-runner/_work/torch-xpu-ops/torch-xpu-ops/torch-xpu-ops/.github/scripts/prepare_pytorch.py", line 33, in run
    return subprocess.run(cmd, check=True, **kwargs)
  File "/opt/python/cp310-cp310/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'checkout', '']' returned non-zero exit status 128.
Error: Process completed with exit code 1.
```

Note that this regression seems to be specific to scheduled events trigger as build stage in this workflow seem to work fine on PR events, for example: https://github.com/intel/torch-xpu-ops/actions/runs/30613713926:
```
+ [[ main == *\h\t\t\p\s\:\/\/* ]]
+ PYTORCH_REPO=https://github.com/pytorch/pytorch.git
+ PYTORCH_COMMIT=main
+ [[ main == *\h\t\t\p\s\:\/\/* ]]
+ TORCH_XPU_OPS_REPO=https://github.com/intel/torch-xpu-ops.git
+ TORCH_XPU_OPS_COMMIT=main
+ python /localdisk/home/jenkins/actions-runner/_work/torch-xpu-ops/torch-xpu-ops/torch-xpu-ops/.github/scripts/prepare_pytorch.py --workspace=/localdisk/home/jenkins/actions-runner/_work/torch-xpu-ops/torch-xpu-ops --pytorch-repo=https://github.com/pytorch/pytorch.git --pytorch-commit=main --torch-xpu-ops-repo=https://github.com/intel/torch-xpu-ops.git --torch-xpu-ops-commit=main --onednn-commit=
+ git clone https://github.com/pytorch/pytorch.git /localdisk/home/jenkins/actions-runner/_work/torch-xpu-ops/torch-xpu-ops/pytorch
Cloning into '/localdisk/home/jenkins/actions-runner/_work/torch-xpu-ops/torch-xpu-ops/pytorch'...
Updating files:  90% (19704/21709)
...
Updating files: 100% (21709/21709), done.
+ git checkout main
```

CC: @mengfei25 @chuanqi129

--- a/.github/workflows/pull_dpclang.yml
+++ b/.github/workflows/pull_dpclang.yml
@@ -107,7 +107,7 @@ jobs:
     with:
       category: dpclang
       runner: build
-      pytorch: ${{ github.base_ref }}
+      pytorch: ${{ github.base_ref || github.ref_name }}
 
   linux-ut:
     needs: build-wheels
@@ -124,6 +124,6 @@ jobs:
     with:
       category: dpclang
       runner: pvc
-      pytorch: ${{ github.base_ref }}
+      pytorch: ${{ github.base_ref || github.ref_name }}
       torch_xpu_ops: ${{ 'main' }}
       ut: ${{ matrix.ut_name }}

